pkg/config/validation/validation.go

```diff
@@ -372,16 +372,26 @@ func ValidateHTTPHeaderOperationName(name string) error {
 	return nil
 }
 
+// Copy from https://github.com/bufbuild/protoc-gen-validate/blob/a65858624dd654f2fb306d6af60f737132986f44/module/checker.go#L18
+var httpHeaderValueRegexp = regexp.MustCompile("^[^\u0000-\u0008\u000A-\u001F\u007F]*$")
+
 // ValidateHTTPHeaderValue validates a header value for Envoy
 // Valid: "foo", "%HOSTNAME%", "100%%", "prefix %HOSTNAME% suffix"
-// Invalid: "abc%123"
+// Invalid: "abc%123", "%START_TIME%%"
 // We don't try to check that what is inside the %% is one of Envoy recognized values, we just prevent invalid config.
 // See: https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers.html#custom-request-response-headers
 func ValidateHTTPHeaderValue(value string) error {
-	if strings.Count(value, "%")%2 != 0 {
-		return errors.New("single % not allowed.  Escape by doubling to %% or encase Envoy variable name in pair of %")
+	if !httpHeaderValueRegexp.MatchString(value) {
+		return fmt.Errorf("header value configuration %s is invalid", value)
 	}
-	return nil
+
+	if err := validateHeaderValue(value); err != nil {
+		return fmt.Errorf("header value configuration %s is invalid", value)
```

Comment:
If the validation function has the exact description of the error why not use it and provide it to the users?

Reply:
updated